### DISCUSSION TITLE
Exposing MAC addess as device connection

### DISCUSSION THREAD

**Discussion by nijel** [make-all/tuya-local#1774] [Ideas]
Home Assistant allows merging devices from multiple integrations. One of the ways to do that is via connections and MAC address. Would it be possible to expose this from tuya-local? 

It would then allow other integrations such as Unifi to add networking info to the device (such as presence or bandwidth usage).

On the Home Assistant side, all what is needed is to add `CONNECTION_NETWORK_MAC` connection to the device registry like shown in https://developers.home-assistant.io/docs/device_registry_index/#manual-registration.

**Comment by make-all** [comment]
The documentation you link to is for registering devices with no connected entities. Such devices are not supported by this integration and are not likely to be in future.  Mac address is a detail specific to lower levels of the TCP/IP stack that is not generally available for integrations that follow the Home Assistant guidelines and use third party libraries to implement network protocols.

**Reply by nijel** [reply]
The documentation link is the first mention of this I could find, there are probably better places. It looks like this in the real integration: https://github.com/home-assistant/core/blob/eb81a4204e1ec32d15519524f93d9dc726298c3e/homeassistant/components/shelly/entity.py#L320-L322

Quickly going through existing integrations that expose this, indeed the MAC address is provided by the third-party library doing the device access. For example, with Shelly, it matches device unique ID.

I don't know Tuya stack enough to know whether it's available somewhere or not, but it would be useful to include in Home Assistant.

**Comment by cvroque** [comment]
After spending a while tracking a lone lightbulb that was not behaving correctly and wishing the device page had the bulb MAC address  I was about to open a new discussion and stumbled upon this one.
I see that many integrations now report the device MAC on their pages while having connected entities, did anything change in HA and could that be possible now?
Here are some examples of native integrations that have this info and connected entities:
<img width="319" height="236" alt="image" src="https://github.com/user-attachments/assets/3d27e294-3620-46f2-9021-ca3735afe667" />
<img width="321" height="238" alt="image" src="https://github.com/user-attachments/assets/77eb7b0d-47e9-4227-8fac-965f56f5d29d" />

**Reply by make-all** [reply]
HA now eavesdrops on DHCP messages on the local network, and it is apparently possible to get the information from the dhcp integration. But this is not well documented, the only documentation is the user level documentation here: https://www.home-assistant.io/integrations/dhcp/ there is no corresponding developer documentation, even the discovery documentation only mentions mDNS, SSDP and getting the network info needed for rolling your own.
